File: Android/app/src/main/java/catglo/com/deliverydroid/settings/Settings_ListOptions.java
package catglo.com.deliverydroid.settings;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import catglo.com.deliveryDatabase.Order;
import catglo.com.deliverydroid.R;
import catglo.com.deliverydroid.homeScreen.HomeScreen_ListFragmentDragDrop;

@SuppressWarnings("unused")
public class Settings_ListOptions extends Activity {
	Order order;
	View sampleTableRowView;
	private ViewGroup prieviewLayout;
	private SharedPreferences sharedPreferences;
	private OnSharedPreferenceChangeListener listener;

	 
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_settings_list_options);
		Intent i = this.getIntent();
		order = (Order) i.getSerializableExtra("order");
	
		if (order==null){
			order = new Order();
			order.address = "1234 State St. New Orleans LA 06890";
			order.cost = 9.99f;
			order.tipTotalsForThisAddress.averagePercentageTip = 10;	
		}
		
		order.tipTotalsForThisAddress.averagePercentageTip = 0.1f;
		order.tipTotalsForThisAddress.averageTip = 3.99f;
		order.tipTotalsForThisAddress.bestTip = 7;
		order.tipTotalsForThisAddress.deliveries = 21;
		order.tipTotalsForThisAddress.lastTip = 0.01f;
		order.tipTotalsForThisAddress.worstTip = 0;
		
		prieviewLayout = (ViewGroup)findViewById(R.id.sampleCellContainer);

        LayoutInflater vi = (LayoutInflater)getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        sampleTableRowView = vi.inflate(R.layout.home_screen_table_row, null);
        prieviewLayout.addView(sampleTableRowView);

        //TODO: Read the prefs and set the values so they show when we launch from home screen

	}

	@Override
	public void onResume(){
		super.onResume();
		 sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
	        listener = new OnSharedPreferenceChangeListener() {public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,String key) {
	        	updatePreview();
			}};
	        sharedPreferences.registerOnSharedPreferenceChangeListener(listener);
	         
	        updatePreview();
	}
	@Override
	public void onPause(){
		super.onPause();
		sharedPreferences.unregisterOnSharedPreferenceChangeListener(listener);
	}
	
	
	
	void updatePreview(){
		HomeScreen_ListFragmentDragDrop.formatTableRowViewForOrder(
                sampleTableRowView,
                order,
                PreferenceManager.getDefaultSharedPreferences(getApplicationContext()),
                1);
	}
	

}

File: Android/app/src/main/java/catglo/com/deliverydroid/settings/SettingsListOptions.kt
package catglo.com.deliverydroid.settings

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.os.Bundle
import android.preference.PreferenceFragment
import android.preference.PreferenceManager


import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import catglo.com.deliveryDatabase.Order
import catglo.com.deliverydroid.R
import catglo.com.deliverydroid.homeScreen.HomeScreen_ListFragmentDragDrop


class SettingsListOptionsFragment : PreferenceFragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.home_list_view_settings)


    }

}

class SettingsListOptions : AppCompatActivity() {
    internal var order: Order? = null
    internal var sampleTableRowView: View? = null
    private var prieviewLayout: ViewGroup? = null
    private var sharedPreferences: SharedPreferences? = null
    private var listener: OnSharedPreferenceChangeListener? = null



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings_list_options)


        val i = this.intent
        order = i.getSerializableExtra("order") as Order?
        if (order == null) {
            order = Order()
            order?.address = "1234 State St. New Orleans LA 06890"
            order?.cost = 9.99f
            order?.tipTotalsForThisAddress?.averagePercentageTip = 10f
        }
        order?.let {
            it.tipTotalsForThisAddress.averagePercentageTip = 0.1f
            it.tipTotalsForThisAddress.averageTip = 3.99f
            it.tipTotalsForThisAddress.bestTip = 7f
            it.tipTotalsForThisAddress.deliveries = 21
            it.tipTotalsForThisAddress.lastTip = 0.01f
            it.tipTotalsForThisAddress.worstTip = 0f
        }


        prieviewLayout = findViewById<View>(R.id.sampleCellContainer) as ViewGroup

        val vi = getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        sampleTableRowView = vi.inflate(R.layout.home_screen_table_row, null)
        prieviewLayout!!.addView(sampleTableRowView)

        //TODO: Read the prefs and set the values so they show when we launch from home screen

    }

    public override fun onResume() {
        super.onResume()
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        listener = OnSharedPreferenceChangeListener { sharedPreferences, key -> updatePreview() }
        sharedPreferences!!.registerOnSharedPreferenceChangeListener(listener)

        updatePreview()
    }

    public override fun onPause() {
        super.onPause()
        sharedPreferences!!.unregisterOnSharedPreferenceChangeListener(listener)
    }


    fun updatePreview() {
        HomeScreen_ListFragmentDragDrop.formatTableRowViewForOrder(
            sampleTableRowView,
            order!!,
            PreferenceManager.getDefaultSharedPreferences(applicationContext),
            1
        )
    }


}
